src/hooks/useDarkMode.ts

```diff
@@ -0,0 +1,18 @@
+import { useState, useEffect } from 'react';
+
+export default function useDarkMode() {
+  const [isDarkMode, setIsDarkMode] = useState(false);
+
+  useEffect(() => {
+    // Check system preference for dark mode with media query
+    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
```

Comment:
i know "window" works fine here, but think of it when we actually deploy this , will it be able to access window then ? maybe we can check if we can access window first , like have some conditions maybe

Reply:
When deployed it can still access window.